msteams-platform/graph-api/proactive-bots-and-messages/graph-proactive-bots-and-messages.md

```diff
@@ -0,0 +1,256 @@
+
+### ✔ [Create](../../bots/how-to/create-a-bot-for-teams.md) and [Publish](../../concepts/deploy-and-publish/overview.md) your [proactive messaging bot](../../concepts/bots/bot-conversations/bots-conv-proactive.md) for Teams
+
+>[!TIP]
+> The production-ready [**Company Communicator**](../..//samples/app-templates.md#company-communicator) app template enables broadcast messaging and is a good foundation for building your proactive bot application.
+
+### ✔ Get the `teamsAppId` for your app
+
+You will need the   `teamsAppId`   for the next steps. It can be retrieved from your organization's app catalog.  
+
+**Microsoft Graph page reference.** [teamsApp resource type](/graph/api/resources/teamsapp?view=graph-rest-1.0)
+
+**HTTP GET** request:
+
+```http
+GET /appCatalogs/teamsApps?$filter=externalId eq '{IdFromManifest}'
+```
+
+The request will return a `teamsCatalogApp`  object. The returned object's `id`  is the app's catalog generated app ID and is different from the id that you provided in your Teams app manifest:
+
+```json
+{
+  "value": [
+    {
+      "id": "b1c5353a-7aca-41b3-830f-27d5218fe0e5",
+      "externalId": "f31b1263-ba99-435a-a679-911d24850d7c",
+      "name": "Test App",
+      "version": "1.0.1",
+      "distributionMethod": "Organization"
+    }
+  ]
+}
+```
+
+### ✔ Determine whether your bot is currently installed for a message recipient
+
+**Microsoft Graph page reference:** [List apps installed for user](/graph/api/user-list-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP GET** request:
+
+```http
+GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsAppDefinition&$filter=teamsAppDefinition/teamsAppId eq '{teamsAppId}'
+```
+
+This request will return an empty array if the app is not installed, or an array with a single [teamsAppInstallation](/graph/api/resources/teamsappinstallation?view=graph-rest-beta) object if it is already installed.
+
+### ✔ Install your app
+
+**Microsoft Graph reference:** [Install app for user](/graph/api/user-add-teamsappinstallation?view=graph-rest-beta&tabs=http)
+
+**HTTP POST** request:
+
+```http
+POST /users/{user-id}/teamwork/installedApps
+{
+   "teamsApp@odata.bind" : "https://graph.microsoft.com/beta/appCatalogs/teamsApps/{teamsAppId}"
+}
+```
+
+If the user has Microsoft Teams running, they may see the app install immediately. Alternatively, a restart may be necessary to see the installed app.
+
+### ✔ Retrieve the conversation **chatId**
+
+When your app is installed for the user, the bot will receive a `conversationUpdate` [event notification](../../resources/bot-v3/bots-notifications.md#team-member-or-bot-addition) that will contain the necessary information to send the proactive message.
+
+The `chatId` can also be retrieved as follows:
+
+**Microsoft Graph reference:** [Get chat](/graph/api/chat-get?view=graph-rest-beta&tabs=http)
```

Comment:
Instead of this which requires a different permission set, we should call out the new API to get the chat id associated with a bot:

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps/{installation-id}/chat 

The {installation-id} will be fetched from the response of 

GET https://graph.microsoft.com/beta/users/{user-id}/teamwork/installedApps?$expand=teamsApp&$filter=teamsApp/id eq '{teamsAppId}' 

We can then add: 
Alternatively, you can also get the chat id from:
GET https://graph.microsoft.com/beta/users/{user-id}/chats?$filter=installedApps/any(a:a/teamsApp/id eq '{teamsAppId}') ; however this needs the permission Chat.Read.All.